### PR TITLE
reportportal: upload all test case results

### PR DESCRIPTION
**Pull request by kkaarreell**
- Upload all test results for a given test case, not just the first one.
- Store guest name and role as test case attributes.
- Rename `get_rp_api()` to `rp_api_get()`
- Add and use `rp_api_post()` and `rp_api_put()`

Fixes issue #3129

Pull Request Checklist

* [ ] implement the feature
* [ ] write the documentation
* [ ] extend the test coverage
* [ ] update the specification
* [ ] adjust plugin docstring
* [ ] modify the json schema
* [ ] mention the version
* [ ] include a release note

### CODE REVIEW

File: tmt/steps/report/reportportal.py [hunk 1]
```diff
@@ -479,111 +479,123 @@ def go(self, *, logger: Optional[tmt.log.Logger] = None) -> None:
...
-                        status = self.TMT_TO_RP_RESULT_STATUS[result.result]
-
-                        # Upload log
+                        # for multi-host tests store also provision name and role
+                        if result.guest.name != 'default-0':
+                            item_attributes.append({'key': 'guest_name', 'value': result.guest.name})
+                        if result.guest.role:
+                            item_attributes.append({'key': 'guest_role', 'value': result.guest.role})
+
+                    if test.contact:
```
Comment (kkaarreell) [inline tmt/steps/report/reportportal.py:504]: FYI, lines below haven't been changed, just shifted.

**Comment by kkaarreell** [comment]
To ease the review pls use
https://app.semanticdiff.com/gh/teemtee/tmt/pull/3135/changes#tmt/steps/report/reportportal.py

**Comment by kkaarreell** [comment]
Any idea why the EPEL9 build failed?

**Review by KwisatzHaderach — APPROVED** [review]

**Comment by thrix** [comment]
@kkaarreell seems this change is not tested at all, I am wondering, if you think there could be way to test this?

**Comment by kkaarreell** [comment]
> @kkaarreell seems this change is not tested at all, I am wondering, if you think there could be way to test this?

So what I did was taking a simple single-host plan and add few other hosts in the `provision` step to make it run on multiple hosts. And then running tmt with all the parameters/variables required for reportportal reporting.

**Review by 4N0body5 — APPROVED** [review]

**Review by therazix — APPROVED** [review]

**Comment by happz** [comment]
Unrelated failures, merging.

**Pull request merged**